src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -49,684 +48,700 @@
+    public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
+    {
+        IEnumerable<FilterLog> GetLogs(IEnumerable<FilterLog> logs, CancellationTokenSource cancellationTokenSource)
         {
-            int id = (int)filterId;
-            FilterType filterType = _blockchainBridge.GetFilterType(id);
-            switch (filterType)
+            using (cancellationTokenSource)
             {
-                case FilterType.BlockFilter:
+                foreach (FilterLog log in logs)
                 {
-                    return _blockchainBridge.FilterExists(id)
-                        ? ResultWrapper<IEnumerable<object>>.Success(_blockchainBridge.GetBlockFilterChanges(id))
-                        : ResultWrapper<IEnumerable<object>>.Fail($"Filter with id: '{filterId}' does not exist.");
-                }
-
-                case FilterType.PendingTransactionFilter:
-                {
-                    return _blockchainBridge.FilterExists(id)
-                        ? ResultWrapper<IEnumerable<object>>.Success(_blockchainBridge
-                            .GetPendingTransactionFilterChanges(id))
-                        : ResultWrapper<IEnumerable<object>>.Fail($"Filter with id: '{filterId}' does not exist.");
-                }
-
-                case FilterType.LogFilter:
-                {
-                    return _blockchainBridge.FilterExists(id)
-                        ? ResultWrapper<IEnumerable<object>>.Success(
-                            _blockchainBridge.GetLogFilterChanges(id).ToArray())
-                        : ResultWrapper<IEnumerable<object>>.Fail($"Filter with id: '{filterId}' does not exist.");
-                }
-
-                default:
-                {
-                    throw new NotSupportedException($"Filter type {filterType} is not supported");
+                    yield return log;
                 }
             }
         }
 
-        public ResultWrapper<IEnumerable<FilterLog>> eth_getFilterLogs(UInt256 filterId)
-        {
-            int id = (int)filterId;
+        // because of lazy evaluation of enumerable, we need to do the validation here first
+        CancellationTokenSource cancellationTokenSource = new(_rpcConfig.Timeout);
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
 
-            return _blockchainBridge.FilterExists(id)
-                ? ResultWrapper<IEnumerable<FilterLog>>.Success(_blockchainBridge.GetFilterLogs(id))
-                : ResultWrapper<IEnumerable<FilterLog>>.Fail($"Filter with id: '{filterId}' does not exist.");
-        }
+        SearchResult<BlockHeader> fromBlockResult;
+        SearchResult<BlockHeader> toBlockResult;
 
-        public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
+        if (filter.FromBlock == filter.ToBlock)
+            fromBlockResult = toBlockResult = _blockFinder.SearchForHeader(filter.ToBlock);
+        else
         {
```

Comment:
Minor-style: One thing that is bothering me and makes it harder to read is the inconsistency of brackets in code blocks in same if-else. if is bracketless and then you have else that is longer block and has {}. Looks weird and inconsistent to me, makes it harder to follow the logic and boundaries at just a glance.